chanzuckerberg/idseq-web
# Make sample report page load last saved viz

# Description

This closes the loop so that a user will see what they expect when they save a report and go back to it from the samples page. The help text is "The current view will be saved as the default for this sample".

![image](https://user-images.githubusercontent.com/28797/52822746-72f60000-3067-11e9-8232-b3a4d097dc75.png)

* Url params will override the saved state
* The saved state will update the URL params so that the saved report can be shared as a URL
* Interaction with cookie state is untouched and unspecified -- we will need to sync on this as discussed
* We keep the whole history of saves but only the last one is ever loaded -- not sure if we want to do anything more about history -- or whether it should work the same for heatmaps
* the last saved report is per sample, so multiple users could overwrite each other's saves -- not sure what is required here

Also, I changed a few lines of the heatmap code to make it more generic. 

## Type of change


# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. load a report page
2. click on taxon tree
3. save 
4. reload page without any url params
5. see saved state
6. click on table view
7. reload with url params
8. see table view

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?


# Checklist:

- [] I have run through the testing script to make sure current functionality is unchanged
